tests/test_exclude.py

```diff
@@ -0,0 +1,61 @@
+import sys
+from subprocess import check_output
+
+
+def test_no_exclude(repo_dir):
+    output = check_output(
```

Comment:
Since here we are only interested in checking the exclude mechanism, not that the command line arguments gets correctly fed to the code, it would be better to call `non_interactive_output` and capture its ouput.

Something like

```
class Capture(list):
    def __enter__(self):
        sys.stdout = StringIO()
        return self
    def __exit__(self, *args):
        self.extend(self._stringio.getvalue().splitlines())
        sys.stdout = sys.__stdout__

with Capture() as output:
    non_interactive_output(...)

assert "file1" in output 
...
```
        
A bit hacky, but that will save us a call to a subprocess (which means faster tests and more accurate coverage reports).

A real clean solution, would be to split the result computation from the _printing_ of this result. That way we could leverage on the _json_format_ option to produce a dict that would be way more convenient to check.

I agree this may go beyond the scope of this PR though.